### ISSUE TITLE
ANR in GooglePlayReceiver with version 0.8.4

### ISSUE THREAD

**Issue by Fishfield** [firebase/firebase-jobdispatcher-android#190]
In Android Vitals I see the following ANR happening with version `0.8.4`
````
"main" tid=1 Blocked 
"main" prio=5 tid=1 Blocked
  | group="main" sCount=1 dsCount=0 obj=0x75127730 self=0xee805400
  | sysTid=4750 nice=-4 cgrp=default sched=0/0 handle=0xf1dcd534
  | state=S schedstat=( 626303126 138569108 748 ) utm=50 stm=12 core=0 HZ=100
  | stack=0xff0f1000-0xff0f3000 stackSize=8MB
  | held mutexes=
  at com.firebase.jobdispatcher.GooglePlayReceiver.ˏ (:288)
- waiting to lock <0x0e5e576b> (a android.support.v4.util.SimpleArrayMap) held by thread 47
  at o.ᖫ.ˊ (:88)
  at com.firebase.jobdispatcher.FirebaseJobDispatcher.ˋ (:1085)
````

I am calling the following method when `onStartJobJob(JobParameters)` is called from the `JobService`

````
	public static void scheduleJob(Context context) {

		FirebaseJobDispatcher dispatcher = new FirebaseJobDispatcher(new GooglePlayDriver(context));

		Job watchDogJob = dispatcher.newJobBuilder()
				.setService(WatchDogAlarmJobService.class)
				.setTag(WATCH_DOG_TAG)
				.setReplaceCurrent(true)
				.setTrigger(Trigger.executionWindow(ALARM_INTERVAL_S, ALARM_INTERVAL_S))
				.build();

		Logger.d(WATCH_DOG_TAG, "scheduleJob: JobDispatcher::mustSchedule");
		dispatcher.mustSchedule(watchDogJob);
	}
````

I know there is already version `0.8.5` out there, but I am not sure if this will fix it. Any ideas, thoughts on that one?

Thanks

**Comment by LouisCAD** [comment]
@Fishfield Did you finally update to version `0.8.5`? Did you see the issue again afterwards?

**Comment by tasneembohra** [comment]
I am still facing this issue on `0.8.5` in oreo devices.

**Comment by mobsome** [comment]
Are you calling some Job Dispatcher methods from your JobService main thread? If yes then 
the root cause is the same as in https://github.com/firebase/firebase-jobdispatcher-android/issues/190.
Please take a look. I pasted a solution for it.

**Comment by Fishfield** [comment]
@LouisCAD sry for the late answer, but yes this is fixed with version `0.8.5` of the library
